retirement-math.py

```diff
@@ -79,34 +282,31 @@ def format_money(val):
     exp = expenses[curr_age - age]
     if curr_age in one_time_expenses: exp += one_time_expenses[curr_age]
 
-    market_return = real_mkt_return()
-    retirement_nw = retirement_nw * market_return
-    retirement_nw += inc
-    retirement_nw -= exp
-
-    cash_case = cash_case + inc - exp
-    cash_case /= 1.02 # real return of -2% on cash portfolio due to inflation
+    # Update the portfolio
+    portfolios.pass_year()
+    portfolios.remove_money(exp)
+    portfolios.add_money(inc)
 
     print('age: ', curr_age, end=';\t')
     # format the numbers
-    print('market_return: ', '{:.2%}'.format((market_return - 1)), end=';\t')
     print('post-tax income: ', format_money(inc), end=';\t')
     print('expenses: ', format_money(exp), end=';\t')
-    print('nw: ', format_money(retirement_nw))
-
-    # These will later be used to plot the net worth over time in millions of dollars
-    age_to_nw[curr_age] = retirement_nw / 1000
-    age_to_all_cash_portfolio[curr_age] = cash_case / 1000
+    print('nw: ', portfolios.get_nw())
+    year += 1
 
-# exit()
 import matplotlib.pyplot as plt
-plt.plot(list(age_to_nw.keys()), list(age_to_nw.values()))
-plt.plot(list(age_to_all_cash_portfolio.keys()), list(age_to_all_cash_portfolio.values()))
-plt.title('Net Worth Over Time')
+# Plot the net worth history of each portfolio over time (in millions)
+#   Use the legend to label each portfolio using the class name
+nw_history = [portfolio.get_nw_history() for portfolio in portfolios.portfolios]
+# normalize the net worth history (shift up index by age): turn to dict
+nw_history = {i + age - 2: [nw_history[j][i] / 1000 for j in range(len(nw_history))] for i in range(len(nw_history[0]))}
```

Comment:
This `-2` offset was added because the graph was misaligned without it. Find the off-by-one error requiring this (or explain why the -2 is justified)